core/commands/inline_diff.py

```diff
@@ -219,26 +219,33 @@ def get_object_contents(self, object_hash):
         """
         return self.git("show", "--no-color", object_hash)
 
+    def get_file_contents_binary(self, file_path):
+        """
+        Given an absolute file path, return the binary contents of that file
+        as a string.
+        """
+        file_path = os.path.join(self.repo_path, file_path)
+        with open(file_path, "rb") as f:
+            return f.read()
+
     def get_file_contents(self, file_path):
         """
         Given an absolute file path, return the text contents of that file
         as a string.
         """
-        file_path = os.path.join(self.repo_path, file_path)
+        binary = self.get_file_contents_binary(file_path)
         try:
-            with open(file_path, "rt", encoding="utf-8") as f:
-                return f.read()
+            return binary.decode('utf-8')
         except UnicodeDecodeError as unicode_err:
-            with open(file_path, "rt", encoding="latin-1") as f:
-                return f.read()
+            return binary.decode('latin-1')
 
     def get_object_from_string(self, string):
         """
         Given a string, pipe the contents of that string to git and have it
         stored in the current repo, and return an object-hash that can be
         used to diff against.
         """
-        stdout = self.git("hash-object", "-w", "--stdin", stdin=string)
+        stdout = self.git("hash-object", "-w", "--stdin", stdin=string, encode=False)
```

Comment:
A question about `hash-object` in common: Each call creates a new blob file in git's object database. This might be fine, if the file is commited anyway. But how do you ensure no longer files are deleted? From my current perspective you create dangling objects this way and polute the database. Even a simple `git gc` does not remove them.